lambdas/backend/express-route-handlers.js

```diff
@@ -492,7 +492,7 @@ async function postAdminCatalogVisibility(req, res) {
         // try {
             let swagger = await exports.apigateway.getExport({
                 restApiId: req.body.apiKey.split('_')[0],
-                stageName: req.body.apiKey.split('_')[1],
+                stageName: req.body.apiKey.substr(req.body.apiKey.indexOf('_') + 1),
```

Comment:
Line 516 should also use this strategy to prevent stages with similar prefixes from having collisions when saved to S3 (e.g., "apiId_prefix_morestuff" would collide with "apiId_prefix_otherstuff"):

```
Key: `catalog/unsubscribable_${req.body.apiKey.split('_')[0]}_${req.body.apiKey.split('_')[1]}.json`,
```

Reply:
Done